src/storage/copier/csv_reader.cpp

```diff
@@ -0,0 +1,515 @@
+    } break;
+    case LogicalTypeID::STRING: {
+        parse_chunk.getValueVector(column)->setValue(columnSizes[column]++, str_val);
+    } break;
+    case LogicalTypeID::DATE: {
+        parse_chunk.getValueVector(column)->setValue(
+            columnSizes[column]++, Date::fromCString(str_val.c_str(), str_val.length()));
+    } break;
+    case LogicalTypeID::TIMESTAMP: {
+        parse_chunk.getValueVector(column)->setValue(
+            columnSizes[column]++, Timestamp::fromCString(str_val.c_str(), str_val.length()));
+    } break;
+    case LogicalTypeID::INTERVAL: {
+        parse_chunk.getValueVector(column)->setValue(
+            columnSizes[column]++, Interval::fromCString(str_val.c_str(), str_val.length()));
+    } break;
+    case LogicalTypeID::VAR_LIST: {
+        auto copyDescription =
+            CopyDescription({filePath}, common::CopyDescription::FileType::CSV, *csvReaderConfig);
+        auto value = TableCopyUtils::getVarListValue(
+            str_val, 1, str_val.length() - 2, type, copyDescription);
+        ValueVector::copyValueToVector(parse_chunk.getValueVector(column)->getData(),
+            parse_chunk.getValueVector(column).get(), value.get());
+        columnSizes[column]++;
+    } break;
+    case LogicalTypeID::FIXED_LIST: {
+        auto copyDescription =
+            CopyDescription({filePath}, common::CopyDescription::FileType::CSV, *csvReaderConfig);
+        auto value = TableCopyUtils::getArrowFixedListVal(
+            str_val, 1, str_val.length() - 2, type, copyDescription);
+        ValueVector::copyValueToVector(parse_chunk.getValueVector(column)->getData(),
+            parse_chunk.getValueVector(column).get() + columnSizes[column], value.get());
+        columnSizes[column]++;
+    } break;
+    default:
+        throw CopyException("Unsupported data type " +
+                            LogicalTypeUtils::dataTypeToString(type.getLogicalTypeID()) +
+                            " inside LIST");
+    }
+    parse_chunk.getValueVector(column)->state->selVector->selectedSize++;
+    // move to the next column
+    column++;
+}
+
+bool BaseCSVReader::AddRow(
+    DataChunk& insert_chunk, column_id_t& column, std::string& error_message, uint64_t buffer_idx) {
+    if (mode == ParserMode::PARSING_HEADER) {
+        return true;
+    }
+    linenr++;
+    if (row_empty) {
+        row_empty = false;
+        if (return_types.size() != 1) {
+            if (mode == ParserMode::PARSING) {
+                // Set This position to be null
+                ;
+            }
+            column = 0;
+            return false;
+        }
+    }
+    if (column < return_types.size()) {
+        throw CopyException(StringUtils::string_format(
+            "Error in file {} on line {}: expected {} values per row, but got {}", filePath, linenr,
+            return_types.size(), column));
+    }
+    if (mode == ParserMode::PARSING &&
+        columnSizes[0] >= common::StorageConstants::NODE_GROUP_SIZE) {
```

Comment:
Are you initializing value vectors to the size of `common::StorageConstants::NODE_GROUP_SIZE` anywhere? Otherwise, by default, if you initialize the value vector, it should be `DEFAULT_VECTOR_CAPACITY`.